addons/payment/views/payment_templates.xml

```diff
@@ -56,14 +56,72 @@
             <t t-set="footer_template_id"
                t-value="footer_template_id or 'payment.footer'"/>
 
+            <!-- === Tokens === -->
+            <t t-if="token_count>0"> 
+                Saved payment methods
+                <div class="card">
+                    <t t-foreach="tokens" t-as="token">
+                        <div name="o_payment_option_card" class="card-body o_payment_option_card">
+                            <label>
+                                <!-- === Radio button === -->
+                                <input name="o_payment_radio"
+                                    type="radio"
+                                    t-att-checked="token.id == default_payment_option_id"
+                                    t-att-data-payment-option-id="token.id"
+                                    t-att-data-provider="token.provider_code"
+                                    data-payment-option-type="token"/>
+                                <!-- === Token name === -->
+                                <span class="payment_option_name" t-esc="token.display_name"/>
+                                <!-- === "V" check mark === -->
+                                <t t-call="payment.verified_token_checkmark"/>
+                                <!-- === "Fees" badge === -->
+                                <span t-if="fees_by_provider.get(token.provider_id)"
+                                    class="badge rounded-pill text-bg-secondary ms-1">
+                                        + <t t-esc="fees_by_provider.get(token.provider_id)"
+                                            t-options="{'widget': 'monetary', 'display_currency': currency}"/>
+                                        Fees
+                                </span>
+                                <!-- === "Test token" badge === -->
+                                <t t-call="payment.test_token_badge"/>
+                                <!-- === "Unpublished" badge === -->
+                                <span t-if="not token.provider_id.is_published" class="badge rounded-pill text-bg-danger ms-1">
+                                    Unpublished
+                                </span>
+                                <div class="ms-4">
+                                    <span class="fst-italic small text-muted">
+                                    Managed by
+                                    <span class="token_payment_provider_name" t-esc="token.provider_id.name"/>
```

Comment:
I saw that the spec ask for the provider code here, instead of the name. Why did you choose to go with the name nonetheless?
Since I also think the name is slightly better, even if it has important downsides, I'll approve the PR. This could be an interesting point to discuss with @dalaOdoo though

Reply:
I don't know if it's a good enough argument, but "code" is starting from lowercase. Also, another case is that if someone would duplicate the provider and give them another name if we would use code instead of name it would still show the same code, which might be a desirable thing depending on what we want actually. But I used name since in the specs it was uppercase.

Reply:
More important: we have the `payment_custom` module. I know, there isn't any token created by this provider, so this is irrelevant, but this is the main argument on why using the name instead of the code makes sense in my opinion. But then, relying on the code instead of the name makes sure the displayed provider name is coherent and not `Stripe Main Account for Company Thingy thing`, as that could be the case if the provider was renamed.

tldr; using the code is making sure the displayed name is correct and not personalized.